unit/java_bytecode/expr2java.cpp

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************\
+
+ Module: Unit tests for expr-to-java string conversion
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <java_bytecode/expr2java.h>
+
+TEST_CASE("0.0 double to string", "[core][expr2java_test]")
```

Comment:
According to the unit test guidelines, I think this should actually be tagged with: `"[core][java_bytecode][expr2java][floating_point_to_java_string]"` 

I'd probably group then all in a 
```
SCENARIO("floating_point_to_java_string", 
  "[core][java_bytecode][expr2java][floating_point_to_java_string]")
```
And then replace each test case with a WHEN("Converting 0.0 double to string")` etc.

Reply:
Sure, but switching to `WHEN` is pointless, since I'm not doing any mockup/teardown before and after each call.